### ISSUE TITLE
[macOS] Dash 18.0.1 signed using 'Apple Distribution' certificate which is not meant for distribution to end-users

### ISSUE THREAD

**Issue by core-code** [dashpay/dash#4982]
previous releases were properly signed using the right ("Developer ID") certificates (first using 'Dash Core Group, Inc.' certificates and then using  'The Dash Foundation').

however the newest release 18.0.1 is improperly signed using a 'Apple Distribution' certificate instead of the right 'Developer ID' certificate.

'Apple Distribution' certificates are only meant for uploading to Apple for software that is going to be distributed to the Mac App Store - its not meant for distributing software to end-users and may present an issue for users that are trying to launch the software.

**Comment by suschizu** [comment]
download https://github.com/dashpay/dash/releases/download/v18.0.2/dashcore-18.0.2-osx.dmg is again signed with a Apple Development signature:
`EXPECTED: 
The Dash Foundation, Inc. (44RJ69WHFF) 
FOUND: 
Apple Development: pasta@dash.org (9LJX49W7UG)`

**Comment by core-code** [comment]
18.0.2 is broken too

**Comment by PastaPastaPasta** [comment]
How so?

**Comment by core-code** [comment]
"Apple Development" certificates are *not* meant for distributing software to end users. they are only meant to upload software to the Mac App Store. your users will not be able to launch DashDash without disabling GateKeeper.

**Comment by PastaPastaPasta** [comment]
Sooo.. It's not supposed to be a "Apple Distribution" or a "Apple Development" but "Developer ID"?

**Comment by PastaPastaPasta** [comment]
Do you know how these "Develop ID" certificates are generated? Inside of Xcode, all that I see are these three options..
<img width="201" alt="image" src="https://user-images.githubusercontent.com/6443210/194918186-67431d3e-d19b-4a72-a330-153654912e5d.png">

A different developer previously handled the macOS signing, but he left the project. So we are trying to figure it out properly now 🙈

**Comment by core-code** [comment]
if you don't have them you should be able to obtain them from Xcode => Settings/Preferences => Accounts => Manage Certificates…

if they are not there and cannot be added....maybe the devforums ( https://developer.apple.com/forums/ ) can help:


<img width="205" alt="Screenshot 2022-10-10 at 19 03 54" src="https://user-images.githubusercontent.com/1261954/194918758-c51db0d1-adf3-48de-a3d8-f3f24636b097.png">

**Comment by core-code** [comment]
i believe proper signing requires a paid apple developer membership, so that may be a reason why they don't show up

**Comment by QuantumExplorer** [comment]
We have a paid membership.

**Comment by PastaPastaPasta** [comment]
Okay, we've got the certificate figured out. I'm building 18.1.0-rc.1 signed Mac binaries right now to see if it works. I'll post a link to them here once done and we can make sure we've got it figured out :D

**Comment by PastaPastaPasta** [comment]
@core-code can you look at https://github.com/dashpay/dash/releases/download/v18.1.0-rc.1/dashcore-18.1.0-rc.1-osx.dmg (my gpg sig here: https://github.com/dashpay/dash/releases/download/v18.1.0-rc.1/dashcore-18.1.0-rc.1-osx.dmg.asc)

**Comment by core-code** [comment]
looks great and confirmed valid - thanks!

```
codesign -dvv /Volumes/Dash-Core/Dash-Qt.app 
Executable=/Volumes/Dash-Core/Dash-Qt.app/Contents/MacOS/Dash-Qt
Identifier=org.dash.Dash-Qt
Format=app bundle with Mach-O thin (x86_64)
CodeDirectory v=20500 size=256764 flags=0x0(none) hashes=8017+3 location=embedded
Signature size=8984
Authority=Developer ID Application: The Dash Foundation, Inc. (44RJ69WHFF)
Authority=Developer ID Certification Authority
Authority=Apple Root CA
Timestamp=10 Oct 2022 at 22:51:54
Info.plist entries=17
TeamIdentifier=44RJ69WHFF
Sealed Resources version=2 rules=13 files=32
Internal requirements count=1 size=112

```

**Comment by PastaPastaPasta** [comment]
I'm gonna close this now, I don't want to replace previously published binaries (don't want two "proper" binaries for one tag). But this should be fixed in the next release (18.1.0 which should come out next week)

**Comment by core-code** [comment]
thanks!
